core/org.openjdk.jmc.flightrecorder/src/main/java/org/openjdk/jmc/flightrecorder/parser/IParserExtension.java

```diff
@@ -69,4 +69,13 @@
 	 * @return a new event sink factory
 	 */
 	IEventSinkFactory getEventSinkFactory(IEventSinkFactory subFactory);
+
+	/**
+	 * creates an instance of IConstantPoolExtension at the beginning of the load of a recording
+	 * 
+	 * @return an instance implementing IConstantPoolExtension
+	 */
+	default IConstantPoolExtension createConstantPoolExtension() {
```

Comment:
Perhaps we might as well make the other methods default too?

		default String getValueInterpretation(String eventTypeId, String fieldId) {
			return null;
		}
		default IEventSinkFactory getEventSinkFactory(IEventSinkFactory subFactory) {
			return subFactory;
		}